src/debug/inc/diagnosticsipc.h

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef __DIAGNOSTICS_IPC_H__
+#define __DIAGNOSTICS_IPC_H__
+
+#include <stdint.h>
+
+#ifdef FEATURE_PAL
+  struct sockaddr_un;
+#else
+  #include <Windows.h>
+#endif /* FEATURE_PAL */
+
+// typedef void (*ErrorCallback)(const char *szMessage, uint32_t code);
```

Comment:
?

Reply:
I will add a callback as input to the constructor to bubble up initialization errors.